### DISCUSSION TITLE
`atomsWithQuery` not updating when `staleTime` is used

### DISCUSSION THREAD

**Discussion by gilmarsquinelato** [pmndrs/jotai#1778] [General]
First of all, thank you for the amazing job you've done!

The situation I found was regarding `atomsWithQuery` when used in conjunction with `staleTime`.
The issue is that the query atom doesn't update its dependencies/components when its data get stale.
An example is when fetching based on a selected user ID, the problem occurs when selecting an already loaded user ID, where the ID atom changes, but the selected user atom doesn't, so the components don't change as well.

I tested with react-query separately and it works perfectly, even if I use it with jotai but not the `atomsWithQuery`, only the atoms to store the selected user id.

The reproduction steps are:
1. select a user (it will show the user details)
2. select a different user (it will change to this user details)
3. select the user from step 1 (it will not change)

The playground to reproduce: https://codesandbox.io/s/fast-https-f2p5g2?file=/src/App.js

Does anyone know why this is happening?
Is it expected or is it a bug?

**Comment by dai-shi** [comment]
Thanks for reporting and a good reproduction!

`staleTime: Infinity` means it should never fetch again, right?
Yeah, this is unexpected.

Can anyone dig into it??

**Reply by gilmarsquinelato** [reply]
Hello @dai-shi 
Happy for helping with the reproduction :D

about `staleTime: Infinity` yep, it means that it should never fetch again.
But, even if you use another value, let's say 5 secs, if you switch from one key to another and come back to the first one in that 5 secs window, it will reproduce the bug.

I have a guess that it might be related to how the integration is caching the values.
I didn't dig into the code, but it looks like the integration is checking if the response is stale, but not the key itself, which is something important :)
